src/cdk/dateadapter/date-adapter.ts

```diff
@@ -10,23 +10,23 @@ import {inject, InjectionToken, LOCALE_ID} from '@angular/core';
 import {Observable, Subject} from 'rxjs';
 
 /** InjectionToken for datepicker that can be used to override default locale code. */
-export const MAT_DATE_LOCALE = new InjectionToken<string>('MAT_DATE_LOCALE', {
+export const DATE_LOCALE = new InjectionToken<string>('DATE_LOCALE', {
```

Comment:
I'm worried that this is easily confused with Angular's own locale token. It might be worthwhile to make this "CDK_DATE_LOCALE" so that it's at least unambiguous.